### PR TITLE
OCR2DR tooling

### PR DESCRIPTION
**Pull request by pinebit**
Closes https://smartcontract-it.atlassian.net/browse/DRO-38

The tooling is inspired by the prior work done by VRF team (kudos!).
All the details can be found in `USAGE.md`.

[DRO-38]: https://smartcontract-it.atlassian.net/browse/DRO-38?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: core/scripts/ocr2dr/templates/oracle.toml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+contractID = "{{oracle_contract_address}}"
+ocrKeyBundleID = "{{ocr2_key_bundle_id}}"
+p2pv2Bootstrappers = [
+  "{{p2p_bootstrapper}}"
+]
+relay = "evm"
+pluginType = "directrequest"
+transmitterID = "{{node_eth_address}}"
+observationSource = """
+    decode_log   [type="ethabidecodelog" abi="OracleRequest(bytes32 requestId, bytes data)" data="$(jobRun.logData)" topics="$(jobRun.logTopics)"]
```
Comment (bolekk) [inline core/scripts/ocr2dr/templates/oracle.toml:15]: Update to  OracleRequest(bytes32 indexed requestId, uint64 subscriptionId, bytes data)

File: core/scripts/ocr2dr/src/generate_jobspecs_cmd.go [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+	for i := 1; i < len(nodes); i++ {
+		oracleLines := replatePlaceholders(lines, *chainID, *p2pPort, *contractAddress, bootHost, bootstrapNode, nca[i])
+		outputPath := filepath.Join(artefactsDir, nodes[i].url.Host+".toml")
+		err = writeLines(oracleLines, outputPath)
+		helpers.PanicErr(err)
+		fmt.Println("Saved oracle node jobspec:", outputPath)
+	}
+}
+
+func replatePlaceholders(lines []string, chainID, p2pPort int64, contractAddress, bootHost string, boot *nodeConfig, node *nodeConfig) (output []string) {
```
Comment (bolekk) [inline core/scripts/ocr2dr/src/generate_jobspecs_cmd.go:62]: typo "replate"

File: core/scripts/ocr2dr/templates/bootstrap.toml [hunk 3]
```diff
@@ -0,0 +1,10 @@
+type = "bootstrap"
+schemaVersion = 1
+name = "OCR2DR bootstrap"
+forwardingAllowed = false
+id = "4"
```
Comment (bolekk) [inline core/scripts/ocr2dr/templates/bootstrap.toml:5]: This is not needed.

File: core/scripts/ocr2dr/templates/bootstrap.toml [hunk 4]
```diff
@@ -0,0 +1,10 @@
+type = "bootstrap"
+schemaVersion = 1
+name = "OCR2DR bootstrap"
```
Comment (bolekk) [inline core/scripts/ocr2dr/templates/bootstrap.toml:3]: Maybe add some unique id or timestamp to job names?

**Review by bolekk — CHANGES_REQUESTED** [review]
Looks great! Just a few minor comments.

File: core/scripts/ocr2dr/templates/oracle.toml [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+    decode_cbor  [type="cborparse" data="$(decode_log.data)"]
+    run_computation    [type="bridge" name="ea_bridge" requestData="{\\"id\\": $(jobSpec.externalJobID), \\"data\\": $(decode_cbor)}" timeout="20s"]
```
Comment (bolekk) [inline core/scripts/ocr2dr/templates/oracle.toml:17]: Also here add:     \\"subscriptionId\\": $(decode_log.subscriptionId)

File: core/scripts/ocr2dr/templates/bootstrap.toml (hunk 4 above)
Comment (pinebit) [reply core/scripts/ocr2dr/templates/bootstrap.toml]: Done

**Review by bolekk — APPROVED** [review]
Thanks!

**Review by KuphJr — APPROVED** [review]

**Pull request merged**